utils/python/CIME/SystemTests/erp.py

```diff
@@ -12,133 +12,50 @@
-            shutil.copy("env_build.xml", os.path.join("LockedFiles","env_build.xml"))
-
-            # update the case to use the new values
-            self._case.read_xml()
-
-            # Use the second executable that was created
-            exeroot = self._case.get_value("EXEROOT")
-            cime_model = CIME.utils.get_model()
-            exefile  = os.path.join(exeroot,"%s.exe"%(cime_model))
-            exefile2 = os.path.join(exeroot,"%s.ERP%d.exe"%(cime_model,run))
-            if (os.path.isfile(exefile)):
-                os.remove(exefile)
-            shutil.copy(exefile2, exefile)
-
-            case_setup(self._case, test_mode=True, reset=True)
-            stop_n      = self._case.get_value("STOP_N")
-            stop_option = self._case.get_value("STOP_OPTION")
-
-            if run == 1:
-                expect(stop_n > 2, "ERROR: stop_n value %d too short"%stop_n)
-                rest_n = stop_n/2 + 1
-                self._case.set_value("REST_N", rest_n)
-                self._case.set_value("REST_OPTION", stop_option)
-                self._case.set_value("HIST_N", stop_n)
-                self._case.set_value("HIST_OPTION", stop_option)
-                self._case.set_value("CONTINUE_RUN", False)
-                suffix = "base"
-            else:
-                rest_n = stop_n/2 + 1
-                stop_new = stop_n - rest_n
-                expect(stop_new > 0, "ERROR: stop_n value %d too short %d %d"%(stop_new,stop_n,rest_n))
-                self._case.set_value("STOP_N", stop_new)
-                self._case.set_value("CONTINUE_RUN", True)
-                self._case.set_value("REST_OPTION","never")
-                suffix = "rest"
-            self.run_indv(suffix=suffix)
-
-        self._component_compare_test("base", "rest")
+        SystemTestsCompareTwo.__init__(self, case, True, run_one_st_archive = True)
+
+    def _case_one_setup(self):
+        pass
+
+    def _case_two_setup(self):
+        """ Case two uses half the number of threads and tasks as the defaults and case one. """
+        for comp in self._case.get_values("COMP_CLASSES"):
+            if comp == "DRV":
+                comp = "CPL"
+            ntasks = self._case.get_value("NTASKS_%s"%comp)
+            if ntasks > 1:
+                rootpe = self._case.get_value("ROOTPE_%s"%comp)
+                self._case.set_value("NTASKS_%s"%comp, ntasks // 2)
+                self._case.set_value("ROOTPE_%s"%comp, rootpe // 2)
+            nthreads = self._case.get_value("NTHRDS_%s"%comp)
+            if nthreads > 1:
+                self._case.set_value("BUILD_THREADED", True)
+                self._case.set_value("NTHRDS_%s"%comp, nthreads // 2)
+
+        rest_n = self._case.get_value("STOP_N") // 2 + 1
+        self._case.set_value("REST_N", rest_n)
+        self._case.set_value("REST_OPTION", self._case.get_value("STOP_OPTION"))
+
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _pre_run_one_hook(self):
```

Comment:
> would it make sense to rename case1 and case2 to something like baseline and test? I believe they were supposed to be interchangeable; but with this test they can't be.

We're not totally consistent about making case1 the 'baseline' - e.g., the PET test turns on threading for case1 and turns it off for case2 - so I'd argue that case2 is more of the 'baseline' here. I feel that the current number-based names illustrate exactly the asymmetry that exists - namely, that case1 is run first, and then case2 is run second. So for tests like this new ERP test that require a certain run to happen first, it seems more clear that that run must be done as case1. I feel that changing the names to 'baseline' and 'test' would actually make this asymmetry _less_ clear. So I'd prefer keeping the names as they are.

In which case, `_inter_baseline_test_run` doesn't make a lot of sense, because we aren't calling the runs 'baseline' and 'test'. I also find this name confusing, because 'baseline' is overloaded here - I think of the BASELINE phase of the test when I see it.

As I think about it more, I still personally find something like `_pre_run_one` or `post_run_two` acceptable and clear.